compressor_integration_tests/src/lib.rs

```diff
@@ -0,0 +1,103 @@
+use openssl::ssl::{SslConnector, SslMethod, SslVerifyMode};
+use postgres::Client;
+use postgres_openssl::MakeTlsConnector;
+use rand::{distributions::Alphanumeric, thread_rng, Rng};
+use std::{borrow::Cow, collections::BTreeMap, fmt};
+
+use synapse_compress_state::StateGroupEntry;
+
+pub mod map_builder;
+
+pub static DB_URL: &str = "postgresql://synapse_user:synapse_pass@localhost/synapse";
+
+pub fn add_contents_to_database(room_id: &str, state_group_map: &BTreeMap<i64, StateGroupEntry>) {
+    // connect to the database
+    let mut builder = SslConnector::builder(SslMethod::tls()).unwrap();
+    builder.set_verify(SslVerifyMode::NONE);
+    let connector = MakeTlsConnector::new(builder.build());
+
+    let mut client = Client::connect(DB_URL, connector).unwrap();
+
+    // build up the query
+    let mut sql = "".to_string();
+
+    for (sg, entry) in state_group_map {
+        // create the entry for state_groups
+        sql.push_str(&format!(
+            "INSERT INTO state_groups (id, room_id, event_id) VALUES ({},{},{});\n",
+            sg,
+            PGEscape(room_id),
+            PGEscape("left_blank")
```

Comment:
Instead of using the slightly awful `PGEscape` here, can't we just do a bunch of `client.execute(..)` instead? It may be a touch slower but I think its much clearer.

Reply:
I used PGEscape because this is the same code used when the compressor generates the SQL (where we do should have PGEscape or at least some sort of safe escaping) - and so if the reader can understand one they can understand both